### ISSUE TITLE
Many missing images on Facebook

### ISSUE THREAD

**Issue by siteadvice** [facebookincubator/Facebook-For-OpenCart#244]
I installed the extension last week and the Opencart product feed is working, but more than half of the products do not have an image in the Facebook shop.
Yesterday I tried pressing the "Resync Products to Facebook" button in the hope that this might force Facebook to re-import the photos, but it has not helped. The same products that had missing photos yesterday still have no photos today.
When I click on the blue "Manage Settings" button in Opencart and go to the "Your Products" tab, I see an animated spinner and the text "Uploading" (screen grab attached), but it has been a day since I initiated the manual upload, so I think all 238 products probably should have been updated by now.
Any suggestions? If I can supply any disgnostics to help, then please let me know.

![Facebook extension screen grab](https://user-images.githubusercontent.com/1247766/64179711-8a59d680-ce5b-11e9-8351-fe5238127320.JPG)

**Comment by shue249** [comment]
@siteadvice What is your website URL?

**Comment by siteadvice** [comment]
https://www.stoveworlduk.co.uk/
https://www.facebook.com/stoveworlduk/


On 4 Sep 2019, at 03:11, shue249 <notifications@github.com<mailto:notifications@github.com>> wrote:


@siteadvice<https://github.com/siteadvice> What is your website URL?

**Comment by shue249** [comment]
@siteadvice 

We looked through several products on your catalog. Here are our findings:

For products whose images are able to load successfully, their image URLs are as follow:
1. Product ID 154, image URL = https://www.stoveworlduk.co.uk/image/catalog/Ottawa-5kw---5-Year-Guarantee.jpg
2. Product ID 220, image URL = http://www.stoveworlduk.co.uk/image/catalog/Ottawa-5kw-Curve---5-Year-Guarantee.jpg
3. Product ID 231, image URL = http://www.stoveworlduk.co.uk/image/catalog/NEW%20Purefire%207.4kw%20defra%20stand%20-%20FRONT.jpg

For products whose images FAIL to load successfully, their image URLs are as follow:
1. Product ID 54, image URL = https://www.stoveworlduk.co.uk/image/catalog/C4.5.jpg
2. Product ID 250, image URL = http://www.stoveworlduk.co.uk/image/catalog/7-8kw%20Curve%20Defra%20Glass.jpg
3. Product ID 166, image URL = https://www.stoveworlduk.co.uk/image/catalog/Ottawa-7-8kw---5-year-Guarantee.jpg

Based on our past experience with other users having image issues, you can look along the lines of
1. Check into the image URL to see if the image is correct
2. Check on the permission of the images, if it has been given read access - https://github.com/facebookincubator/Facebook-For-OpenCart/issues/87

**Comment by shue249** [comment]
@siteadvice As we are using the API to update the product, the status information on the "Your products" tab is not applicable.

On the image problems, what are the image URLs used to display on the web store for those 2 products?

**Comment by siteadvice** [comment]
I have now solved this problem and you might find the solution helpful to other people - the problem was that the domain's DNS and SSL certificate are provided by Cloudflare and the SSL certificate settings on the account were set to "Flexible".

Apparently it is not usual for this to result in infinite 301 redirect loops. I have now switched the Cloudflare settings to "Full (strict)" and subsequently the Facebook shop is fully populated with product photos.

Further information can be found about this phenomenon hre in case it's useful to any of your other users:
https://support.cloudflare.com/hc/en-us/articles/115000219871-Why-does-Flexible-SSL-cause-a-redirect-loop-
